sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedPlainValuesReader.java

```diff
@@ -96,6 +124,33 @@ public final void readLongs(int total, WritableColumnVector c, int rowId) {
     }
   }
 
+  // A fork of `readLongs` to rebase the timestamp values. For performance reasons, this method
+  // iterates the values twice: check if we need to rebase first, then go to the optimized branch
+  // if rebase is not needed.
+  @Override
+  public final void readLongsWithRebase(int total, WritableColumnVector c, int rowId) {
+    int requiredBytes = total * 8;
+    ByteBuffer buffer = getBuffer(requiredBytes);
+    boolean rebase = false;
+    for (int i = 0; i < total; i += 1) {
```

Comment:
The byte code of the loop is:
```
    LINENUMBER 136 L6
    ILOAD 6
    ALOAD 5
    ALOAD 5
    INVOKEVIRTUAL java/nio/ByteBuffer.position ()I
    ILOAD 7
    BIPUSH 8
    IMUL
    IADD
    INVOKEVIRTUAL java/nio/ByteBuffer.getLong (I)J
    INVOKESTATIC org/apache/spark/sql/catalyst/util/RebaseDateTime.lastSwitchJulianTs ()J
    LCMP
    IFGE L7
    ICONST_1
    GOTO L8
```
We could avoid mul like 
```java
    int pos = buffer.position();
    int endPos = pos + total * 8;
    long threshold = RebaseDateTime.lastSwitchJulianTs();
    while (pos < endPos) {
      rebase |= buffer.getLong(pos) < threshold;
      pos += 8;
    }
```
Would it be faster?

Reply:
I tried it and the perf has no difference.